src/components/FullscreenModal/Footer/useStickyFooter.ts

```diff
@@ -0,0 +1,75 @@
+import { useCallback, useEffect, useState } from 'react';
+import debounce from 'lodash.debounce';
+
+export const useStickyFooter = (
+  modalRef: React.MutableRefObject<HTMLElement>,
+  modalFooterRef: React.MutableRefObject<HTMLElement>,
+): { isFixed: boolean; shouldShowScrollToTopButton: boolean } => {
+  const [isFixed, setIsFixed] = useState(false);
+  const [
+    shouldShowScrollToTopButton,
+    setShouldShowScrollToTopButton,
+  ] = useState(false);
+  const showScrollToTopButton = useCallback(() => {
+    if (modalRef.current === null) return;
+    const isScrollable =
+      modalRef.current.scrollHeight > modalRef.current.offsetHeight;
+    setShouldShowScrollToTopButton(isScrollable);
+    setIsFixed(false);
+  }, [modalRef, setShouldShowScrollToTopButton]);
```

Comment:
Turns out depending on refs isn't working as well, since ref is a mutable object, which exists for whole lifecycle of the component

Reply:
any suggestion?

Reply:
don't depend on it, just remove it from the list of dependencies. If you need to perform update whenever dom element changes, there is something called callbacks refs: https://reactjs.org/docs/refs-and-the-dom.html#callback-refs

Reply:
But actually you can simply use empty array for dependencies, or don't use `useCallback` here at all. Everything that `showScrollToTopButton` operates on is guaranteed to be the same between renders 🤷